src/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,341 @@
+  simpleValue: boolean,
+  /** the function that gets used to render the content of an option group */
+  optionGroupRenderer: OptionGroupRendererHandler
+  /** the function that can be used to override the default drop-down list of options */
+  menuRenderer: MenuRendererHandler
+  /** the field name to group by */
+  groupByKey: string
+  /** whether the menu should be rendered on top or bottom when it's open */
+  menuPosition: 'top' | 'bottom'
+};
+
+export const Props = {
+  value: t.maybe(t.union([t.Number, t.String, t.Object, t.list(t.Object)])),
+  onChange: t.maybe(t.Function),
+  onValueClick: t.maybe(t.Function),
+  options: t.list(t.Object),
+  size: t.maybe(t.enums.of(['medium', 'small'])),
+  disabled: t.maybe(t.Boolean),
+  searchable: t.maybe(t.Boolean),
+  clearable: t.maybe(t.Boolean),
+  backspaceRemoves: t.maybe(t.Boolean),
+  multi: t.maybe(t.Boolean),
+  flat: t.maybe(t.Boolean),
+  autoBlur: t.maybe(t.Boolean),
+  simpleValue: t.maybe(t.Boolean),
+  menuPosition: t.maybe(t.enums.of(['top', 'bottom'])),
+  menuRenderer: t.maybe(t.Function),
+  groupByKey: t.maybe(t.String),
+  optionGroupRenderer: t.maybe(t.Function),
+  placeholder: t.maybe(t.union([t.String, ReactElement])),
+  noResultsText: t.maybe(t.String),
+  allowCreate: t.maybe(t.Boolean),
+  addLabelText: t.maybe(t.String),
+  valueRenderer: t.maybe(t.Function),
+  optionRenderer: t.maybe(t.Function),
+  delimiter: t.maybe(t.String),
+  onInputChange: t.maybe(t.Function),
+  onFocus: t.maybe(t.Function),
+  onBlur: t.maybe(t.Function),
+  onBlurResetsInput: t.maybe(t.Boolean),
+  onCloseResetsInput: t.maybe(t.Boolean),
+  isLoading: t.maybe(t.Boolean),
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+};
+
+export type Props = RequiredProps & Partial<DefaultProps>
+
+type DefaultedProps = RequiredProps & DefaultProps;
+
+@props(Props, { strict: true })
+export default class Dropdown extends React.Component<Props> {
+
+  static defaultProps: DefaultProps = {
+    delimiter: ',',
+    size: 'medium',
+    disabled: false,
+    searchable: false,
+    clearable: false,
+    multi: false,
+    flat: false,
+    autoBlur: true,
+    simpleValue: true,
+    groupByKey: 'optionGroup',
+    optionGroupRenderer: defaultOptionGroupRenderer,
+    menuRenderer: defaultMenuRenderer,
+    menuPosition: 'bottom'
+  }
+
+  componentDidMount() {
+    this.logWarnings();
+  }
+
+  logWarnings = () => {
+    if (this.props.children) {
+      warn('You\'re passing children. Not expected behaviour');
+    }
+  };
+
+  valueToOption = (value: Props['value'], options: Select.Options) => {
+    if (t.String.is(value) || t.Number.is(value)) {
+      const { multi, delimiter } = this.props as DefaultedProps;
+      if (multi) {
+        const values = String(value).split(delimiter);
+        return values.map(v => find(options, { value: v })!); // TODO: `!`
+      }
+      return find(options, { value })!; // TODO: `!`
```

Comment:
we were already implicitly assuming `value` as a valid option here, not sure how to fix this gracefully